bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeApi.java

```diff
@@ -75,14 +77,34 @@ public SenecHomeResponse getStatistics()
         Request request = httpClient.newRequest(location);
         request.header(HttpHeader.ACCEPT, MimeTypes.Type.APPLICATION_JSON.asString());
         request.header(HttpHeader.CONTENT_TYPE, MimeTypes.Type.FORM_ENCODED.asString());
-        ContentResponse response = request.method(HttpMethod.POST)
-                .content(new StringContentProvider(gson.toJson(new SenecHomeResponse()))).send();
-
-        if (response.getStatus() == HttpStatus.OK_200) {
-            return Objects.requireNonNull(gson.fromJson(response.getContentAsString(), SenecHomeResponse.class));
-        } else {
-            logger.trace("Got unexpected response code {}", response.getStatus());
-            throw new IOException("Got unexpected response code " + response.getStatus());
+        ContentResponse response = null;
+        try {
+            response = request.method(HttpMethod.POST)
+                    .content(new StringContentProvider(gson.toJson(new SenecHomeResponse()))).send();
+            if (response.getStatus() == HttpStatus.OK_200) {
+                return Objects.requireNonNull(gson.fromJson(response.getContentAsString(), SenecHomeResponse.class));
+            } else {
+                logger.trace("Got unexpected response code {}", response.getStatus());
+                throw new IOException("Got unexpected response code " + response.getStatus());
+            }
+        } catch (MalformedJsonException | JsonSyntaxException | InterruptedException | TimeoutException
+                | ExecutionException e) {
+            logger.trace("Issue with getting SenecHomeResponse");
+            logger.trace("location: {}", location);
+            logger.trace("request: {}", request.toString());
+            logger.trace("request.getHeaders: {}", request.getHeaders());
+            if (response == null) {
+                logger.trace("response is null");
+            } else {
+                logger.trace("response: {}", response.toString());
+                logger.trace("response.getHeaders: {}", response.getHeaders());
+                if (response.getContent() == null) {
+                    logger.trace("response.getContent is null");
+                } else {
+                    logger.trace("response.getContentAsString: {}", response.getContentAsString());
+                }
+            }
+            throw e;
```

Comment:
This change is optional. But in case of an error it will help to understand what is wrong with the response from the SENEC device